homeassistant/components/harmony/switch.py

```diff
@@ -20,6 +21,15 @@ async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up harmony activity switches."""
+    async_create_issue(
+        hass,
+        DOMAIN,
+        "deprecated_switches",
+        breaks_in_ha_version="2023.8.0",
+        is_fixable=False,
+        severity=IssueSeverity.WARNING,
+        translation_key="deprecated_switches",
+    )
```

Comment:
We can tell them which scripts and automations need to be updated

```
            entity_automations = automations_with_entity(hass, entity.entity_id)
            entity_scripts = scripts_with_entity(hass, entity.entity_id)
            if entity_automations:
                automations[entity.entity_id] = entity_automations
            if entity_scripts:
                scripts[entity.entity_id] = entity_scripts

```

Reply:
Nice! Was that always available since issues were a thing or is this relatively new?

Reply:
This also only applies if the entity is referenced and not the device. Can we detect those?